docs/docs/dev_docs/tutorials/token_portal/setup.md

```diff
@@ -0,0 +1,185 @@
+    "strict": true,
+    "declaration": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "downlevelIteration": true,
+    "inlineSourceMap": true,
+    "declarationMap": true,
+    "importHelpers": true,
+    "resolveJsonModule": true,
+    "composite": true,
+    "skipLibCheck": true
+  },
+  "include": [
+    "packages/src/**/*",
+    "contracts/**/*.json",
+    "packages/src/**/*",
+    "packages/aztec-contracts/token_bridge/target/*.json"
+  ],
+  "exclude": ["node_modules", "**/*.spec.ts", "contracts/**/*.ts"],
+  "references": []
+}
+```
+
+# Create a nargo project
+
+Now inside `aztec-token-portal` create a new directory called `aztec-contracts`
+
+Inside `aztec-contracts`, create a nargo contract project by running
+
+```bash
+nargo new -contract token_bridge
+```
+
+Your structure will look something like this
+
+```
+aztec-contracts
+└── token_bridge
+    ├── Nargo.toml
+    ├── src
+				├── main.nr
+```
+
+Inside `Nargo.toml` you will need to add some dependencies. Put this at the bottom:
+
+```json
+[dependencies]
+aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/aztec" }
+value_note = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/value-note"}
+safe_math = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/safe-math"}
+```
+
+We will also need some utility functions that we won’t write in this tutorial. In the `src` directory, create a new file called `token_interface.nr` and paste the contents of [this file](https://github.com/AztecProtocol/aztec-packages/blob/891c1362160693f69bc6a843c7c41776b51b9f7c/yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/token_interface.nr#L4).
+
+Then in the same place create a new file called `[util.nr](http://util.nr)` and paste the contents of [this file.](https://github.com/AztecProtocol/aztec-packages/blob/891c1362160693f69bc6a843c7c41776b51b9f7c/yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/util.nr)
+
+# Create a hardhat project
+
+In the root dir `aztec-token-portal`, create a new directory called `l1-contracts` and `cd` into it. Inside here run `npx hardhat init` and keep hitting enter so you get the default setup.
+
+Once you have a hardhat project set up, delete the `contracts` folder inside `l1-contracts`. We will be downloading a new one in the next step.
+
+## Download Ethereum contracts
+
+We will write the `TokenPortal.sol` contract in this tutorial, but it has many imports that we will need to have locally.
+
+To make this easier we have a standalone repo with all the smart contracts with relative paths - [find it here](https://github.com/catmcgee/contracts). You can either clone this directly into `l1-contracts` (recommended to then `rm -rf .git`), download & unzip, or copy & paste.
```

Comment:
nice idea! Maybe link to the devrel repo instead?

Also add a TODO in the code:
```
<-- TODO(#2453) redirect people to the l1-contracts repo -->
```

https://github.com/AztecProtocol/aztec-packages/issues/2453 -> will be doing this for ease of use xD

Reply:
need it to just be easily checked out (right now) so it needs to be a submodule. it exists in devrels repo now but still links to mine. i think it's fine for now